packages/doc/src/index.ts

```diff
@@ -0,0 +1,92 @@
+import generate from "@babel/generator";
+import { parse } from "@babel/parser";
+import dprint from "dprint-node";
+import { readFile, writeFile } from "fs/promises";
+import type { Config, DbMetadata } from "joist-codegen";
+import pLimit from "p-limit";
+import { Cache } from "./Cache";
+import { CommentStore } from "./CommentStore";
+import { MarkdownCommentStore } from "./MarkdownCommentStore";
+import { IntegrationHandler } from "./integrationHandler";
+import { entityIntegration } from "./integrations/entity";
+import { entityCodegenIntegration } from "./integrations/entityCodegen";
+import { enumIntegration } from "./integrations/enum";
+import { hashString } from "./utils";
+
+class JoistDoc {
+  private cache = new Cache();
+
+  constructor(private commentStore: CommentStore, private metadata: DbMetadata, private config: Config) {}
+
+  /**
+   * Runs an integration handler, handling the common stuff
+   *
+   * - calculates source hash
+   * - asks commentStore for its hash
+   * - if nothing has changed and we have a restore, exit early and use that
+   *
+   * - otherwise: parse, run handler, generated, dprint, write, update cache
+   */
+  async run<T>(integration: IntegrationHandler<T>, topic: T) {
+    const filePath = integration.file(topic, this.config);
+    const file = await readFile(filePath, { encoding: "utf-8" });
```

Comment:
Maybe call this `contents`? I briefly thought `file` was still the file path, i.e. when reading `hashString(file)`.